### PR TITLE
[Diffusion][NPU] Add support for Hunyuan3D

### PR DESCRIPTION
**Pull request by e-martirosian**
## Motivation
This PR adds NPU support to the Hunyuan3D pipeline.

## Modifications

- Added support for building a custom rasterizer for CPU only.
- Set to use CPU in `rasterize `when input tensors are on NPU.
- Converted `image_tensors` to `float32` (previously used `double`) in `_run_delight` - `double` is not supported on NPU.
- Fixed device propagation in `MeshRender`.

### GPU

Before:

gpu_before

After:

gpu_after

### NPU

Before։ pipeline failed.

After:

npu_after

### GPU

The performance difference is within the error margin.

Devices: 1 x Nvidia A10 
Command: `sglang generate --model-path tencent/Hunyuan3D-2 --image-path ./assets/demo.png`

Before: _546.98 seconds_
After: _545.57 seconds_

### NPU

Devices: one chip of Ascend A3
Command: `sglang generate --model-path tencent/Hunyuan3D-2 --image-path ./assets/demo.png`

Generated in _518.22 seconds_

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py [hunk 1]
```diff
@@ -49,16 +48,24 @@ def rasterize(
...
-        clamp_depth = torch.zeros(0, device=pos.device)
+        clamp_depth = torch.zeros(0, device=device)
 
     # pos should be [N, 4], remove batch dim if present
     if pos.dim() == 3:
         pos = pos[0]
 
     findices, barycentric = kernel.rasterize_image(
-        pos, tri, clamp_depth, resolution[1], resolution[0], 1e-6, use_depth_prior
+        pos.to(device), tri.to(device), clamp_depth.to(device), resolution[1], resolution[0], 1e-6, use_depth_prior
```
Comment (VDV1985) [inline python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py:63]: Need to check "ASYNC" copy performance for "pos" and "tri". It probably can be faster if done in parallel.

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py [hunk 2]
```diff
@@ -49,16 +48,24 @@ def rasterize(
...
     )
+
+    if real_device.type != "cuda":
+        findices = findices.to(real_device)
```
Comment (VDV1985) [inline python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py]: Need to check "ASYNC" copy performance for "findices" and "barycentric". It probably can be faster if done in parallel.

File: python/sglang/multimodal_gen/configs/pipeline_configs/base.py [hunk 3]
```diff
@@ -226,6 +226,9 @@ def postprocess_image(self, image):
     # Wan2.2 TI2V parameters
     boundary_ratio: float | None = None
 
+    # Execute rasterization on cpu (Hunyuan3D)
+    rasterization_on_cpu: bool = False
```
Comment (ssshinigami) [inline python/sglang/multimodal_gen/configs/pipeline_configs/base.py]: i suggest to remove server argument and make it by default on CPU if is_npu

**Review by ssshinigami — CHANGES_REQUESTED** [review]

File: python/sglang/multimodal_gen/configs/pipeline_configs/base.py (hunk 3 above)
Comment (e-martirosian) [reply python/sglang/multimodal_gen/configs/pipeline_configs/base.py]: ok

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py (hunk 1 above)
Comment (e-martirosian) [reply python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py:63]: We will work on enabling the rasterizer to run on NPU in the future, so these copy operations will not be needed.

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py (hunk 2 above)
Comment (e-martirosian) [reply python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py]: We will work on enabling the rasterizer to run on NPU in the future, so these copy operations will not be needed.

**Review by ssshinigami — APPROVED** [review]
LGTM

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/hunyuan3d_paint.py [hunk 4]
```diff
@@ -335,11 +335,11 @@ def _run_delight(self, image):
             image_array[:, :, 3] = alpha_channel
             image = PILImage.fromarray(image_array)
 
-            image_tensor = torch.tensor(np.array(image) / 255.0).to(self.device)
+            image_tensor = torch.tensor(np.array(image) / 255.0).float().to(self.device)
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/hunyuan3d_paint.py:338]: maybe we can use `.to(dtype=torch.float, device=self.device)` here

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py [hunk 5]
```diff
@@ -47,7 +50,8 @@ def rasterize(
     use_depth_prior: int = 0,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
     """Rasterize mesh to get face indices and barycentric coordinates."""
-    kernel = _load_custom_rasterizer()
+    device = "cpu" if pos.device.type == "npu" else pos.device.type
+    kernel = _load_custom_rasterizer(device == "cuda")
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py:53]: will it also work for other hardware backends such as AMD?

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/hunyuan3d_paint.py (hunk 4 above)
Comment (e-martirosian) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/hunyuan3d_paint.py:338]: If you strongly prefer this coding style, I can change it, but the current version is more compact in my opinion.

File: python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py (hunk 5 above)
Comment (e-martirosian) [reply python/sglang/multimodal_gen/csrc/render/hunyuan3d_rasterizer/__init__.py:53]: We only check for NPU and run the custom kernel on CPU. This is intentional — we don’t expect it to work on other backends. Developers supporting other hardware can decide to implement this custom kernel for their backend or just run this part on CPU. Otherwise, they might not even notice this part to optimize it.

**Review by mickqian — APPROVED** [review]
